addons/website_crm_partner_assign/controllers/main.py

```diff
@@ -144,6 +140,9 @@ def portal_my_opportunities(self, page=1, date_begin=None, date_end=None, sortby
         # content according to pager
         opportunities = CrmLead.search(domain, order=order, limit=self._items_per_page, offset=pager['offset'])
 
+        if filterby:
+            opportunities = opportunities.sudo().filtered_domain(searchbar_filters[filterby]['domain'])
```

Comment:
Actually this won't do as opp_count won't be accurate, and because of "paged" search, we will return an inconsistent number of records, maybe even empty pages.